### PR TITLE
Search handler for mobile app

### PR DESCRIPTION
**Pull request by YuriHechter**

### CODE REVIEW

File: api/models.go [hunk 1]
```diff
@@ -188,9 +188,30 @@ type MobileContentUnitResponse struct {
 	Items []*MobileContentUnitResponseItem `json:"items"`
 }
 
+type MobileSearchResponse struct {
+	Total int                         `json:"total"`
```
Comment (davgur) [inline api/models.go:192]: Why you use int and int32 and not int64?

File: api/models.go [hunk 2]
```diff
@@ -188,9 +188,30 @@ type MobileContentUnitResponse struct {
...
+	Items []*MobileSearchResponseItem `json:"items"`
+}
+
+type MobileSearchResponseItem struct {
+	ContentUnitUid *string    `json:"contentUnitId,omitempty"`
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply api/models.go:197]: for source results, we don't send content unit id, only source id

File: api/handlers.go [hunk 3]
```diff
@@ -1143,6 +1161,297 @@ func SearchStatsHandler(c *gin.Context) {
...
+
+		for _, hit := range res.SearchResult.Hits.Hits {
+
+			var result es.Result
+			if hit.Source == nil {
+				search.LogIfDeb(&query, fmt.Sprintf("Empty source in hit: %+v.", hit))
+				continue
+			}
+			json.Unmarshal(*hit.Source, &result)
+			if err != nil {
```
Comment (davgur) [inline api/handlers.go:1320]: you check wrong error

File: api/handlers.go [hunk 4]
```diff
@@ -1143,6 +1161,297 @@ func SearchStatsHandler(c *gin.Context) {
...
+					isArticle := firstUnit != nil && cacheM.SearchStats().IsContentUnitTypeArticle(*firstUnit)
+					if isArticle {
+						// Articles collection is not currently supported in mobile
+						search.LogIfDeb(&query, "Skip result for mobile search: Articles Collection.")
+						continue
+					}
+					if firstUnit != nil {
+						// We retrieve collection image according to the first content unit of the collection.
+						// Maybe we should retrieve collection image in a different way.
+						imageStr := fmt.Sprintf(imagesUrlTemplate, *firstUnit)
```
Comment (davgur) [inline api/handlers.go:1363]: client cant to do it?

Comment (YuriHechter) [reply api/handlers.go:1363]: Mobile client is "dummy", he expect to receive everything prepared for him. That is how we send image on other mobile handlers.

File: cache/search_stats.go [hunk 5]
```diff
@@ -159,6 +163,18 @@ func (ssc *SearchStatsCacheImpl) GetProgramByCollectionAndPosition(collection_ui
 	return nil
 }
 
+func (ssc *SearchStatsCacheImpl) GetCollectionFirstUnit(collection_uid string) *string {
```
Comment (davgur) [inline cache/search_stats.go:166]: why point to string?

File: cache/search_stats.go [hunk 6]
```diff
@@ -366,3 +390,47 @@ func (ssc *SearchStatsCacheImpl) loadProgramsByCollectionAndPosition() (map[stri
 	}
 	return ret, nil
 }
+
+func (ssc *SearchStatsCacheImpl) loadCollectionsFirstUnit() (map[string]string, error) {
+	query := `SELECT DISTINCT ON (c.id) c.uid AS collection_uid, cu.uid AS first_content_unit_uid
+	FROM collections AS c
+	JOIN collections_content_units AS ccu ON c.id = ccu.collection_id
+	JOIN content_units AS cu ON ccu.content_unit_id = cu.id
```
Comment (davgur) [inline cache/search_stats.go:398]: you are not check publishe/sequre here

File: api/models.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (davgur) [reply api/models.go:197]: I mean why you use *string / *int32

File: api/models.go (hunk 1 above)
Comment (YuriHechter) [reply api/models.go:192]: fixed

File: api/models.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (YuriHechter) [reply api/models.go:197]: allow this values to be nullable

File: api/handlers.go (hunk 3 above)
Comment (YuriHechter) [reply api/handlers.go:1320]: fixed

File: api/handlers.go [hunk 7]
```diff
@@ -1143,6 +1161,297 @@ func SearchStatsHandler(c *gin.Context) {
 	c.JSON(http.StatusOK, res)
 }
 
+func MobileSearchHandler(c *gin.Context) {
```
[the remark this answers is not in the document]
Comment (YuriHechter) [reply api/handlers.go:1164]: done

File: api/models.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (davgur) [reply api/models.go:197]: usually we use null.String/null.int

**Pull request merged**